content/en/blog/2024/otel-errors/index.md

```diff
@@ -0,0 +1,247 @@
+Here is a trace view for the service [py-otel-server](https://github.com/avillela/otel-errors-talk/blob/main/src/python/server.py). As you can see below, the error spans show up as red dots:
+
+![List of traces in the Jaeger UI](Jaeger01.png)
+
+And if we drill down and zero in on the error span, we see this:
+
+![Attributes and other metadata for an error span in Jaeger](Jaeger02.png)
+
+The span is clearly marked as error, and includes a span event with the exception 
+captured. Jaeger expresses the span event as a log, but does not visualize logs 
+outside of spans. 
+
+### Proprietary backends 
+
+If you’ve been using a proprietary agent to monitor your applications and have 
+recently migrated to OTel, you might notice that an OTel error may not be 
+expressed the way you expect in your observability backend, as compared to the 
+same error captured by the proprietary agent. This is most likely due to the 
+fact that OTel simply models errors differently than how vendors have been 
+modeling them. 
+
+As a broad example, vendors might have their own notion of what constitutes a 
+logical unit of work in an application. You may be familiar with the term 
+`transaction`, which means something slightly different from vendor to vendor. 
+In OTel, this is represented by a span. You’ve likely noticed differences in 
+your data visualization experiences as vendors make their own adjustments to 
+their platforms to accommodate OTLP as a first-class citizen data type. 
+
+As a more specific example, OTel’s notion of span kinds may affect how your OTel 
+error is expressed in your backend. For instance, if you have a trace that has 
+one exception and it’s on an internal span with its status set to `Error`, you 
+should see the trace marked with an error, but it may not be counted toward your 
+overall app error rate. This is because the vendor might have an opinion that 
+only errors on entry point spans (server spans) and consumer spans should be 
+counted toward your error rate. 
+
+If your backend supports trace and log correlation, you should be able to 
+navigate to the associated trace from the log, and vice versa. Furthermore, 
+while Jaeger visualizes span events as logs, some vendors might synthesize span 
+events as its own data type instead of as a log data type, which would impact 
+the way you query that data.
+
+## Conclusion
+Error handling is a challenging yet essential aspect of software development, and 
+OTel offers a comprehensive solution for navigating its complexities. By 
+leveraging OTel's capabilities to record errors through logs and spans and to 
+enhance spans with metadata, you can gain deeper insights into your applications' 
+behavior and more effectively troubleshoot issues. You'll be better equipped to 
+build and maintain resilient, reliable, and high-performing software applications 
+in today's dynamic and demanding environments. 
```

Comment:
This sounds more like an intro. Could you recap what the reader just went through, reminding which key aspects should be considered and what the desired outcome is?

Reply:
Done!